Frameworks/CoreGraphics/CGImage.mm

```diff
@@ -21,981 +21,660 @@
-                    FAIL_FAST_HR(E_UNEXPECTED);
-                    break;
-            }
-            rowOut += xStrideOut;
-            rowStart += xStrideImg;
-        }
-        pImgData += yStrideImg;
-        png_write_row(png_ptr, pRow);
-    }
-    IwFree(pRow);
-    pImage->Backing()->ReleaseImageData();
-
-    png_write_end(png_ptr, NULL);
-    png_destroy_write_struct(&png_ptr, &info_ptr);
-
-    return ret;
+/**
+ @Status Stub
+ */
+CGImageRef CGImageCreateWithMask(CGImageRef image, CGImageRef mask) {
+  // Given how masks are applied during rendering via D2D, we will hold onto the
+  // mask then apply it at the appropriate time.
+  RETURN_NULL_IF(!image);
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
-@interface _UIImageWriterCallback : NSObject
-- (void)image:(UIImage*)image didFinishSavingWithError:(NSError*)err contextInfo:(void*)contextInfo;
-@end
-
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-const CGFloat* CGImageGetDecode(CGImageRef image) {
-    UNIMPLEMENTED();
-    return StubReturn();
+CGImageAlphaInfo CGImageGetAlphaInfo(CGImageRef img) {
+  if (!img) {
+    return kCGImageAlphaNone;
+  }
+  // We would only have three values.
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-CGColorRenderingIntent CGImageGetRenderingIntent(CGImageRef image) {
-    UNIMPLEMENTED();
-    return StubReturn();
+size_t CGImageGetBitsPerPixel(CGImageRef img) {
+  RETURN_RESULT_IF_NULL(img, 0);
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-bool CGImageGetShouldInterpolate(CGImageRef image) {
-    UNIMPLEMENTED();
-    return StubReturn();
+size_t CGImageGetBitsPerComponent(CGImageRef img) {
+  RETURN_RESULT_IF_NULL(img, 0);
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
 /**
- @Status Stub
- @Notes
+ @Status Interoperable
 */
-CFTypeID CGImageGetTypeID() {
-    UNIMPLEMENTED();
-    return StubReturn();
+size_t CGImageGetBytesPerRow(CGImageRef img) {
+  RETURN_RESULT_IF_NULL(img, 0);
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
 /**
  @Status Stub
  @Notes
 */
-bool CGImageIsMask(CGImageRef image) {
-    UNIMPLEMENTED();
-    return StubReturn();
+const CGFloat *CGImageGetDecode(CGImageRef image) {
+  RETURN_NULL_IF(!image);
+  UNIMPLEMENTED();
+  return StubReturn();
 }
 
 /**
  @Status Stub
  @Notes
 */
-CGImageRef CGImageCreateWithMaskingColors(CGImageRef image, const CGFloat* components) {
-    UNIMPLEMENTED();
-    return StubReturn();
-}
\ No newline at end of file
+CGImageRef CGImageCreateWithMaskingColors(CGImageRef image,
+                                          const CGFloat *components) {
+  RETURN_NULL_IF(!image);
+  UNIMPLEMENTED();
+  return StubReturn();
+}
+
+#pragma region WIC_HELPERS
+
+CGImageRef _CGImageGetImageFromData(void *data, int length) {
+  unsigned char *bytes = static_cast<unsigned char *>(data);
+
+  // TODO: (The below code is not really needed.)
```

Comment:
>TODO: (The below code is not really needed.) [](start = 5, length = 44)

Wic should detect the image types, so lets get rid of this code given your TODO.